RIOT-OS/RIOT
# drivers/periph_common: build periph module selectively

### Contribution description

This PR is poc to optimize slightly the build of common periph driver code: in master, all common *.c are built unconditionally (but internally guarded with preprocessor). With this PR, they are only added to the build if the corresponding periph module belong to the build. This saves a few unnecessary called to gcc, for the same result.

The Makefile code could be improved (e.g. factorized) though.




### Testing procedure

A green Murdock should be enough.

To see the effect of this PR, build `examples/hello-world` with `QUIET=0`, on master, the output will contains build for `periph_common/i2c.c`, `periph_common/spi.c`, etc. With this PR, they are gone.




### Issues/PRs references
